wiki/Oppia-Android-Code-Coverage.md

```diff
@@ -0,0 +1,375 @@
+## Table of Contents
+
+- [Overview](#overview)
+- [Understanding Code Coverage](#understanding-code-coverage)
+- [Why is Code Coverage Important?](#why-is-code-coverage-important)
+- [How to use the code coverage tool?](#how-to-use-the-code-coverage-tool)
+    - [Continuous Itegration Checks on Pull Request](#1-continuous-integration-checks-on-pull-requests)
+        - [Understanding the CI Coverage Report](#11-understanding-the-ci-coverage-report)
+    - [Local Command Line Interface Tools](#2-local-command-line-interface-cli-tools)
+        - [Understanding the Html Reports](#21-understanding-the-ci-coverage-report)
+- [Increasing Code Coverage Metrics](#increasing-code-coverage-metrics)
+- [Limitations of the code coverage tool](#limitations-of-the-code-coverage-tool)
+
+# Overview
+In software engineering, code coverage, also called test coverage, is a percentage measure of the degree to which the source code of a program is executed when a particular test suite is run. A program with high code coverage has more of its source code executed during testing, which suggests it has a lower chance of containing undetected software bugs compared to a program with low code coverage.
+
+# Understanding code coverage
+Code coverage measures the extent to which the code is tested by the automated tests. It indicates whether all parts of the code are examined to ensure they function correctly.
+
+Let's take a look at how code coverage works with a simple Kotlin function.
+
+Consider a Kotlin function designed to determine if a number is positive or negative:
+
+```kotlin
+fun checkSign(n: Int): String {
+    return if (n >= 0) {
+        "Positive" 
+    } else { 
+        "Negative"
+    }
+}
+```
+
+A test is created to verify that this function works correctly:
+Please refer to the [Oppia Android testing documentation](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing) to learn more about writing tests.
+
+```
+import org.junit.Test
+import kotlin.test.assertEquals
+
+class checkSignTest {
+    @Test
+    fun testCheckSign_withPositiveInteger_returnsPositive() {
+        assertEquals("Positive", checkSign(4))
+    }
+}
+```
+
+This test checks whether passing a positive integer '+4' `checkSign(4)` correctly returns "Positive" as output.
+However, it doesn’t test how the function handles negative numbers. This means that only a portion of the function’s behavior is being tested, leaving the handling of negative numbers unverified.
+
+For thorough testing, the tests should also check how the function responds to negative inputs, this can be done by adding a test case by passing a negative integer to expect a return of "Negative" as output.
+
+```diff
+import org.junit.Test
+import kotlin.test.assertEquals
+
+class checkSignTest {
+    @Test
+    fun testCheckSign_withPositiveInteger_returnsPositive() {
+        assertEquals("Positive", checkSign(4))
```

Comment:
Here & above: we should use Truth assertThat rather than assertEquals (to avoid being inconsistent with tests seen throughout the codebsae). This is different than the test cases where you use assertEquals in generated code since this is an actual reference that may be copied during development.

Reply:
Ah Got it. Replaced `assertEquals` with `assertThat`